autogluon/task/tabular_prediction/predictor.py

```diff
@@ -63,13 +64,14 @@ def __init__(self, learner):
             To access any trainer method `func()` from this `Predictor`, use: `predictor._trainer.func()`.
         """
         self._learner: Learner = learner  # Learner object
+        self._learner.persist_trainer(low_memory=True)
         self._trainer: AbstractTrainer = self._learner.load_trainer()  # Trainer object
         self.output_directory = self._learner.path
         self.problem_type = self._learner.problem_type
         self.eval_metric = self._learner.objective_func
         self.label_column = self._learner.label
         self.feature_types = self._trainer.feature_types_metadata
-        self.model_names = self._trainer.get_model_names_all()
+        self.model_names = self._trainer.get_model_names_all()  # TODO: Will be outdated if new models are trained through refit_full()
```

Comment:
shouldn't this be updated by `refit_full()`? otherwise things might appear broken to user?

Reply:
Added, clarified that the comment was towards a more general statement that self.model_names should probably be a function to ensure it is always aligned (It doesn't make sense to enable user to mutate it as a variable)

Reply:
switching to function sounds good to me